brave/brave-ios
# Fix #7443: Migrate User Assets from BraveCore to CoreData

## Summary of Changes
We are now storing User's Assets in CoreData instead of in BraveCore. In this way we can fix the problem that auto-discovery could bring back user deleted asset.
There will be some migrations happen when user previously added some custom assets and changed some visibility in older version. 


This pull request fixes #7443

## Submitter Checklist:


## Test Plan:


1. Create a new wallet account and check if all the default visible assets are listed in Portfolio and NFT tab

1. Restore a wallet and check if all the default visible assets are listed in Portfolio and NFT tab

1. On a old version(earlier than 1.53.x), login to wallet and make some changes on visibility for some assets
2. Upgrade to 1.53.x or this branch. 
3. Log into wallet again to see if correct visible assets are still being listed in Portfolio and NFT tab

1. Check if change visibility via Edit Visible Asset is still working 
2. Check adding custom asset is still working
3. Check removing custom asset is still working

1. login or restore a wallet account
2. enable NFT discovery via the popup in NFT tab or the settings screen
3. wait for the discovery timer in BraveCore (sometimes you need to lock the wallet and come back, sometimes you need to relaunch the wallet)
4. check if the NFTs you owned have appeared inside NFT tab
5. Hide one or more NFTs via Edit Visible Assets 
6. Close and relaunch wallet 
7. Check if the hidden NFT comes back inside NFT tab


## Screenshots:



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
